bigframes/display/anywidget.py

```diff
@@ -98,8 +102,10 @@ def __init__(self, dataframe: bigframes.dataframe.DataFrame):
         # there are multiple pages and show "page 1 of many" in this case
         self._reset_batches_for_new_page_size()
         if self._batches is None or self._batches.total_rows is None:
+            # TODO(b/428238610): We could still end up with a None here if the
+            # underlying execution doesn't produce a total row count.
             self._error_message = "Could not determine total row count. Data might be unavailable or an error occurred."
-            self.row_count = 0
+            self.row_count = None
```

Comment:
Let's add another TODO to cheaply discover if we have exactly 1 page. I suspect there are cases where total rows is not set, but there are no additional pages.